control-plane/grpc/proto/pool/v1/pool.proto

```diff
@@ -0,0 +1,157 @@
+syntax = "proto3";
+
+import "misc/common.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+package pool;
+
+// A Mayastor Storage Pool
+// It may have a spec which is the specification provided by the creator
+// It may have a state if such state is retrieved from a mayastor storage node
+message Pool {
+  // Desired specification of the pool and metadata
+  optional PoolDefinition definition = 1;
+  // Runtime state of the pool.
+  optional PoolState state = 2;
+}
+
+// Multiple pools
+message Pools {
+  repeated Pool pools = 1;
+}
+
+// Error type for the control-plane specific errors
+message ReplyError {
+  uint32 status_code = 1;
+  string details = 2;
+}
+
+// Desired spec and the control plane related data
+message PoolDefinition {
+  // Client specification of the pool
+  PoolSpec spec = 1;
+  // Control plane related data for the pool
+  Metadata metadata = 2;
+}
+
+// uuid and the status of pool for the control plane use
+message Metadata {
+  // uuid of the pool
+  google.protobuf.StringValue uuid = 2;
+  // spec status of the pool
+  common.SpecStatus status = 4;
+}
+
+// User specification of a pool.
+message PoolSpec {
+  // id of the mayastor instance
+  string node_id = 1;
+  // id of the pool
+  string pool_id = 2;
+  // absolute disk paths claimed by the pool
+  repeated string disks = 3;
+  // labels to be set on the pool
+  optional common.StringMapValue labels = 5;
+}
+
+// Pool information
+message PoolState {
+  // id of the mayastor instance
+  string node_id = 1;
+  // id of the pool
+  string pool_id = 2;
+  // absolute disk paths claimed by the pool
+  repeated string disks_uri = 3;
+  // current status of the pool
+  PoolStatus status = 4;
+  // size of the pool in bytes
+  uint64 capacity = 5;
+  // used bytes from the pool
+  uint64 used = 6;
+}
+
+// status of the pool
+enum PoolStatus {
+  // unknown state
+  Unknown = 0;
+  // the pool is in normal working order
+  Online = 1;
+  // the pool has experienced a failure but can still function
+  Degraded = 2;
+  // the pool is completely inaccessible
+  Faulted = 3;
+}
+
+// Filter by Node id
+message NodeFilter{
+  string node_id = 1;
+}
+
+// Filter by Pool id
+message PoolFilter{
+  string pool_id = 1;
+}
+
+// Filter by Node and Pool id
+message NodePoolFilter{
+  string node_id = 1;
+  string pool_id = 2;
+}
+
+// Get all the pools from based on the args
```

Comment:
Nit: Maybe this should say something like "Get all pools based on the filter criteria"